presto-native-execution/presto_cpp/main/connectors/arrow_flight/tests/ArrowFlightConnectorAuthTest.cpp

```diff
@@ -0,0 +1,236 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <arrow/testing/gtest_util.h>
+#include <folly/init/Init.h>
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+#include "presto_cpp/main/connectors/arrow_flight/ArrowFlightConnector.h"
+#include "presto_cpp/main/connectors/arrow_flight/Macros.h"
+#include "presto_cpp/main/connectors/arrow_flight/tests/utils/ArrowFlightConnectorTestBase.h"
+#include "presto_cpp/main/connectors/arrow_flight/tests/utils/ArrowFlightPlanBuilder.h"
+#include "presto_cpp/main/connectors/arrow_flight/tests/utils/Utils.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/common/config/Config.h"
+#include "velox/exec/tests/utils/AssertQueryBuilder.h"
+
+using namespace arrow;
+using namespace facebook::velox;
+using namespace facebook::velox::exec::test;
+
+namespace facebook::presto::test {
+
+class TestingServerMiddlewareFactory : public flight::ServerMiddlewareFactory {
+ public:
+  static constexpr const char* kAuthHeader = "authorization";
+  static constexpr const char* kAuthToken = "Bearer 1234";
+  static constexpr const char* kAuthTokenUnauthorized = "Bearer 2112";
+
+  arrow::Status StartCall(
+      const flight::CallInfo& info,
+      const flight::ServerCallContext& context,
+      std::shared_ptr<flight::ServerMiddleware>* middleware) override {
+    auto iter = context.incoming_headers().find(kAuthHeader);
+
+    if (iter == context.incoming_headers().end()) {
+      return flight::MakeFlightError(
+          flight::FlightStatusCode::Unauthenticated,
+          "Authorization token not provided");
+    } else {
+      std::lock_guard<std::mutex> l(mutex_);
+      checkedTokens_.emplace_back(iter->second);
+    }
+
+    if (kAuthToken != iter->second) {
+      return flight::MakeFlightError(
+          flight::FlightStatusCode::Unauthorized,
+          "Authorization token is invalid");
+    }
+
+    return arrow::Status::OK();
+  }
+
+  bool isTokenChecked(const std::string& authToken) {
+    {
+      std::lock_guard<std::mutex> l(mutex_);
+      return std::find(
+                 checkedTokens_.begin(), checkedTokens_.end(), authToken) !=
+          checkedTokens_.end();
+    }
+  }
+
+ private:
+  std::string validToken_;
+  std::vector<std::string> checkedTokens_;
+  std::mutex mutex_;
+};
+
+class TestingAuthenticator : public Authenticator {
```

Comment:
Nit : Names TestAuthenticator and TestAuthenticatorFactory are better than usage of verbs like Testing in the name.